src/java_binding/src/lib.rs

```diff
@@ -425,6 +425,44 @@ pub extern "system" fn Java_com_risingwave_java_binding_Binding_rowGetStringValu
     })
 }
 
+#[no_mangle]
+pub extern "system" fn Java_com_risingwave_java_binding_Binding_rowGetTimestampValue<'a>(
+    env: EnvParam<'a>,
+    pointer: Pointer<'a, JavaBindingRow>,
+    idx: jint,
+) -> JObject<'a> {
+    execute_and_catch(env, move || {
+        let millis = pointer
+            .as_ref()
+            .get_datetime(idx as usize)
+            .0
+            .timestamp_millis();
+        let date_class = env.find_class("java/sql/Timestamp")?;
+        let constructor = env.get_method_id(date_class, "<init>", "(J)V")?;
+        let date_obj = env.new_object_unchecked(date_class, constructor, &[millis.into()])?;
+
+        Ok(date_obj)
+    })
+}
+
+#[no_mangle]
+pub extern "system" fn Java_com_risingwave_java_binding_Binding_rowGetDecimalValue<'a>(
+    env: EnvParam<'a>,
+    pointer: Pointer<'a, JavaBindingRow>,
+    idx: jint,
+) -> JObject<'a> {
+    execute_and_catch(env, move || {
+        let value = pointer.as_ref().get_decimal(idx as usize).to_string();
+        let string_value = env.new_string(value)?;
+        let decimal_class = env.find_class("java/math/BigDecimal")?;
```

Comment:
I am not sure whether we can avoid building the constructor in every call. We may try storing the constructor when we build the row and then reuse the constructor.

Reply:
It's hard to keep a reference of JVM Object and fullfill the memory safety requirements of Rust. Since `JClass<'local>` has lifetime requirement and we can not keep it global. And we can not keep it in scoped storage also since there's no scope in FFI call.

Reply:
I think Java or JVM itself do the same ( find a class and use its constructor to create an object) when creating a BigDecimal object.

Reply:
> so how about the overhead for a constructor? we do it in every chunk or every row?

We do it to every row cell if it is  decimal type.